### ISSUE TITLE
Run DataStreamMonitoring in a separate process

### ISSUE THREAD

**Issue by rizar** [bartvm/blocks#553]
In theory monitoring can be often done in parallel with training on a CPU. I practice I struggle to see how it could be implemented, but maybe somebody wants to think about it.

**Comment by EderSantana** [comment]
Did they solve the problem of how to translate a Theano graph and variables from GPU to CPU? I saw that you guys were having a conversation about that at another issue. The extension could load the graph from checkpoint or get it directly from GPU memory.

**Comment by bartvm** [comment]
Nope, https://github.com/Theano/Theano/pull/2339 is still an open PR. I think transferring the shared variables works, but there's some issue with transferring the optimized graph itself, I think (not sure).

**Comment by rizar** [comment]
Clarification:  https://github.com/Theano/Theano/pull/2339 is about transferring shared variables only. Don't know about the status of the issue in general.
